protocol/src/main/java/org/geysermc/mcprotocollib/network/Flag.java

```diff
@@ -0,0 +1,7 @@
+package org.geysermc.mcprotocollib.network;
+
+public record Flag<T>(String key, Class<T> type) {
+    public T cast(Object obj) {
+        return type.cast(obj);
+    }
+}
```

Comment:
Can we instead make this a class and not expose the `Class<T> type` so that generics could be introduced in the future without breaking